dask/array/linalg.py

```diff
@@ -328,3 +328,69 @@ def svd(a, name=None):
     dask.array.linalg.tsqr: Actual implementation with citation
     """
     return tsqr(a, name, compute_svd=True)
+
+
+def solve_triangular(a, b, lower=False):
+    """
+    Solve the equation `a x = b` for `x`, assuming a is a triangular matrix.
+
+    Parameters
+    ----------
+    a : (M, M) array_like
+        A triangular matrix
+    b : (M,) array_like
+        Right-hand side matrix in `a x = b`
+    lower : bool, optional
+        Use only data contained in the lower triangle of `a`.
+        Default is to use upper triangle.
+
+    Returns
+    -------
+    x : (M,) array
+        Solution to the system `a x = b`. Shape of return matches `b`.
+    """
+
+    import scipy.linalg
+
+    if a.ndim != 2:
+        raise ValueError('a must be 2 dimensional')
+    if b.ndim == 1:
+        if a.shape[1] != b.shape[0]:
+            raise ValueError('a.shape[1] and b.shape[0] must be equal')
+        if a.chunks[1][0] != b.chunks[0][0]:
+            msg = ('a.chunks[1] and b.chunks[0] must be equal. '
+                   'Use .rechunk method to change the size of chunks.')
+            raise ValueError(msg)
+    elif b.ndim == 2:
+        raise NotImplementedError
+    else:
+        raise ValueError('b must be 1 or 2 dimensional')
+
+    nchunks = len(a.chunks[1])
+    token = tokenize(a, b, lower)
+    name = 'solve-triangular-' + token
+
+    def solve_triangular_lower(a, b):
+        return scipy.linalg.solve_triangular(a, b, lower=True)
```

Comment:
Can we move functions like this outside of the `solve_triangular` function?  Dynamically created functions are expensive to serialize.  This is important for distributed computation.